File: app/src/main/java/org/breezyweather/common/utils/helpers/SnackbarHelper.java
package org.breezyweather.common.utils.helpers;

import android.view.View;

import androidx.annotation.Nullable;

import org.breezyweather.BreezyWeather;
import org.breezyweather.common.basic.GeoActivity;
import org.breezyweather.common.snackbar.Snackbar;
import org.breezyweather.common.snackbar.SnackbarContainer;

public class SnackbarHelper {

    public static void showSnackbar(String content) {
        showSnackbar(content, null, null);
    }

    public static void showSnackbar(GeoActivity activity, String content) {
        showSnackbar(activity, content, null, null);
    }

    public static void showSnackbar(String content,
                                    @Nullable String action, @Nullable View.OnClickListener l) {
        showSnackbar(content, action, l, null);
    }

    public static void showSnackbar(GeoActivity activity, String content,
                                    @Nullable String action, @Nullable View.OnClickListener l) {
        showSnackbar(activity, content, action, l, null);
    }

    public static void showSnackbar(String content,
                                    @Nullable String action, @Nullable View.OnClickListener l,
                                    @Nullable Snackbar.Callback callback) {

        GeoActivity activity = BreezyWeather.getInstance().getTopActivity();
        if (activity != null) {
            showSnackbar(activity, content, action, l, callback);
        }
    }

    public static void showSnackbar(GeoActivity activity, String content,
                                    @Nullable String action, @Nullable View.OnClickListener l,
                                    @Nullable Snackbar.Callback callback) {
        if (action != null && l == null) {
            throw new RuntimeException("Must send a non null listener as parameter.");
        }

        if (callback == null) {
            callback = new Snackbar.Callback();
        }

        SnackbarContainer container = activity.provideSnackbarContainer();

        Snackbar.make(container.getContainer(), content, Snackbar.LENGTH_LONG, container.getCardStyle())
                .setAction(action, l)
                .setCallback(callback)
                .show();
    }
}

File: app/src/main/java/org/breezyweather/common/utils/helpers/SnackbarHelper.kt
/**
 * This file is part of Breezy Weather.
 *
 * Breezy Weather is free software: you can redistribute it and/or modify it
 * under the terms of the GNU Lesser General Public License as published by the
 * Free Software Foundation, version 3 of the License.
 *
 * Breezy Weather is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
 * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public
 * License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with Breezy Weather. If not, see <https://www.gnu.org/licenses/>.
 */

package org.breezyweather.common.utils.helpers

import android.view.View
import org.breezyweather.BreezyWeather
import org.breezyweather.common.basic.GeoActivity
import org.breezyweather.common.snackbar.Snackbar

object SnackbarHelper {

    fun showSnackbar(
        content: String,
        action: String? = null,
        activity: GeoActivity? = null,
        listener: View.OnClickListener? = null,
    ) {
        if (action != null && listener == null) {
            throw RuntimeException("Must send a non null listener as parameter.")
        }
        val container = (activity ?: BreezyWeather.instance.topActivity ?: return).provideSnackbarContainer()
        Snackbar.make(container.container, content, Snackbar.LENGTH_LONG, container.cardStyle)
            .setAction(action, listener)
            .setCallback(Snackbar.Callback())
            .show()
    }
}
